parquet/parquet.cpp

```diff
@@ -1356,7 +1356,7 @@ static void add_null(pq_column *c, int16_t r, int16_t d)
     ss << "column '" << c->n->name << "' is required";
     throw pq::ParquetException(ss.str());
   }
-  update_levels(c, r, d);
+  update_levels(c, r, d - 1 ? d > 0 : 0);
```

Comment:
See the example in https://static.googleusercontent.com/media/research.google.com/en//pubs/archive/36632.pdf.  The example output is broken in this PR.